Feature/server/169 scenario management

YIllmann opened the pull request.

simohlsen commented:
Would be great to make the changes to the database merge-able asap so new issues can be implemented on the latest state of the database and wont have to be potentially migrated.

YIllmann commented:
First of: sry for the lasting PR. The changes that were caused of this PR are massive. Thank you for your help an patience. Since your two PRs the following changes were made and require further review:
- added files in web_api: all files are very lookalike due to similar requirements. possible updates due FE changes might follow outside of this PR.
- models: It appears that we both missed the references for vehicles to location in our PlayersToVehicle Table. Therefore i added a column and extended the Unique Constraint to ensure valid data. Luckily the test-data you created represent a special case of mapping :).
- test_action only contains a little test i used with my debugger to ensure the media is updated correctly
- test_scenario does the same for our execution-scenario-vehicle problem.

simohlsen commented on server/scenario/web_api/data/action.py:
```diff
@@ -0,0 +1,185 @@
+        "name": action.name,
+        "min_role": min_role.name if min_role else None,
+        "duration_secs": action.duration_secs,
+        "media_refs": json.loads(action.media_refs) if action.media_refs else [{}],
```
Why `[{}]` not `[]`?

simohlsen commented on server/scenario/web_api/data/action.py:
```diff
@@ -0,0 +1,185 @@
+
+@web_api.post("/action")
+def create_action():
+    """ Creates an empty Action and returns the dbo as JSON. """
+    action = models.Action(
+        name="Neue Maßnahmen",
+        duration_secs=0,
+        required_power=0,
+        media_refs=[{}],
+        result=""
+    )
+    db.session.add(action)
+    db.session.commit()
```
What if you hit the new action button twice without editing the first or is this not possible?

YIllmann replied:
the action id autoincrements and you get 2 action with the same name displayed. no collision on db-side. further, you will be redirected in a different edit window, so you cant create 2 actions. Locations and resource behave the same

simohlsen commented on server/scenario/web_api/data/location.py:
```diff
@@ -0,0 +1,194 @@
+def create_location():
+    location = models.Location(
+        name="Neue Location",
+        media="[{}]",
```
Why `[{}]` not `[]` or just `null`? Also, shouldn't this be `media_refs`?

simohlsen commented on server/scenario/web_api/data/resource.py:
```diff
@@ -0,0 +1,106 @@
+        name="Neue Resource",
+        consumable="False",
+        quantity=10000,
+        media_refs="[{}]"
```
`[]` or `null`
